app/assets/javascripts/uploadAdditionalInformation.js

```diff
@@ -0,0 +1,84 @@
+$(document).ready(function(){
+	Dropzone.autoDiscover = false;
+
+	// Source: https://stackoverflow.com/questions/105034/create-guid-uuid-in-javascript#answer-2117523
+	function uuidv4() {
+		return ([1e7]+-1e3+-4e3+-8e3+-1e11).replace(/[018]/g, c =>
+			(c ^ crypto.getRandomValues(new Uint8Array(1))[0] & 15 >> c / 4).toString(16)
+		)
+	}
+
+	function setUploadUrl(url) {
+		uploadAdditionalInfoDropzone.options.url = url;
+	}
+
+	function prepareAwsHiddenInputs(responseData) {
+		$('#aws_key').attr('name', 'key').val(responseData.fields['key']);
```

Comment:
This could be done as a loop - every field in the 'fields' object has an input element created.  The advantage being that if in the future, aws added some other field to the gem to output in this form, you would not need to change anything - and neither would I in the API as I have made it dynamic

Reply:
I think I would need to change something as I'd still need to add the raw field into the HTML file? This only takes the new data and populates pre-existing, but empty, input fields.